pkg/kapis/devops/v1alpha3/webhook/workflow_handler.go

```diff
@@ -0,0 +1,140 @@
+/*
+Copyright 2022 The KubeSphere Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package webhook
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"k8s.io/klog"
+	"kubesphere.io/devops/pkg/api/devops/v1alpha3"
+	"kubesphere.io/devops/pkg/event/models/workflowrun"
+	"kubesphere.io/devops/pkg/kapis/devops/v1alpha3/pipelinerun"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+type pipelineRunIdentifier struct {
+	namespaceName string
+	pipelineName  string
+	scmRefName    string
+	buildNumber   string
+}
+
+func (identifier *pipelineRunIdentifier) toIdentifier() string {
```

Comment:
How about renaming this to `String()`.

For example: `func (identifier *pipelineRunIdentifier) String() string {`

the struct name already is an `xxIdentifier`, it makes me feel confused about `toIdentifier`. It already is `identifier`. do we need turn to `identifier` again?